frontend/controller/service-worker.js

```diff
@@ -204,11 +216,17 @@ sbp('sbp/selectors/register', {
         let subID = null
         // get a real push subscription only if both browser permissions allow and user wants us to
         if (notificationEnabled && granted) {
+          const subscriptionOptions = await sbp('push/getSubscriptionOptions')
           subscription = await registration.pushManager.getSubscription()
+          if (subscription && !bufferEq(subscription.options.applicationServerKey, subscriptionOptions?.applicationServerKey)) {
```

Comment:
With the way this is written, is it possible for `subscription.options.applicationServerKey` to be an array and for `subscriptionOptions?.applicationServerKey` to be undefined/null?

If so, the `bufferEq` implementation won't work on this.

Reply:
I see what you mean. They way things work now, it shouldn't really be possible, but technically it's possible. It makes the check uglier, but I'll fix it.